src/main/java/org/broadinstitute/hellbender/engine/spark/AssemblyRegionArgumentCollection.java

```diff
@@ -24,19 +25,51 @@
     public static final int DEFAULT_MAX_READS_PER_ALIGNMENT = 50;
     public static final double DEFAULT_ACTIVE_PROB_THRESHOLD = 0.002;
     public static final int DEFAULT_MAX_PROB_PROPAGATION_DISTANCE = 50;
+    public static final String INDEL_PADDING_LONG_NAME = "padding-around-indels";
+    public static final String SNP_PADDING_LONG_NAME = "padding-around-snps";
+
+    /**
+     * The following parameters can be confusing due to the overlap between active regions, assembly regions, and genotyping regions,
+     * all of which are implemented as the {@link org.broadinstitute.hellbender.engine.AssemblyRegion} class.
+     *
+     * An active region is a genomic interval in which we call variants.  In order to output phased calls we try to put nearby variants
+     * in the same active region.  The size of active regions are bounded by the {@code minAssemblyRegionSize} and {@code maxAssemblyRegionSize}
+     * parameters, and within those bounds are determined by the variants present and the {@code activeProbThreshold} and {@code maxProbPropagationDistance}.
+     * Importantly active regions are excusively a matter of delegating responsibility for calling and have nothing to do with the size
+     * of the span over which we perform local assembly and pair-HMM.
+     *
+     * An assembly region is the padded interval surrounding an active region over which we perform local assembly.  Padding is useful to phase with
+     * any variation that may be just outside the active region, to avoid dangling ends in the assembly region, and to resolve indels.  The
+     * {@code assemblyRegionPadding} parameter determines the number of extra bases as assembly region contains on either side of the active region
+     * it surrounds.
+     *
+     * A genotyping region is an interval surrounding an active region in which we perform pair-HMM and Smith-Waterman alignment.  Even though
+     * we ultimately only genotype and call variants within the active region, we call entire haplotypes as an intermediate step, hence the
+     * needed for an expanded genotyping region.  The parameters {@code snpPaddingForGenotyping} and {@code indelPaddingForGenotyping} determine
```

Comment:
I think mentioning the defaults here (snp genotyping padding = 25, indel genotyping padding = 75, assembly region padding = 100) would be helpful for the reader (i.e. me)

Reply:
I'm nervous about hard-coding numbers -- what if they change? -- but I replaced the `{@code }`s with `{@link }`s so the reader can quickly navigate to the defaults in the IDE.